agent/pkg/tunnel/proxy/proxy.go

```diff
@@ -0,0 +1,190 @@
+package proxy
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/libp2p/go-msgio/protoio"
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/edgemesh/agent/pkg/tunnel/controller"
+	"github.com/kubeedge/edgemesh/agent/pkg/tunnel/proxy/pb"
+	"github.com/kubeedge/edgemesh/common/constants"
+)
+
+const (
+	MaxReadSize  = 4096
+	MaxRetryTime = 3
+)
+
+var ProxyProtocol protocol.ID = "/libp2p/edgemesh/1.0.0"
```

Comment:
the protocol name should indicate what the protocol does， such as `/libp2p/circuit/relay/0.1.0` for circuit relay

Reply:
fix